packages/gatsby-plugin-static-image/src/babel-plugin-parse-static-images.ts

```diff
@@ -0,0 +1,137 @@
+import * as types from "@babel/types"
+import { PluginObj } from "@babel/core"
+import { hashOptions, evaluateImageAttributes } from "./utils"
+import fs from "fs-extra"
+import path from "path"
+import {
+  StringLiteral,
+  NumericLiteral,
+  BooleanLiteral,
+  ArrayExpression,
+  TemplateLiteral,
+  ObjectProperty,
+} from "@babel/types"
+
+type AttrType =
+  | StringLiteral
+  | NumericLiteral
+  | BooleanLiteral
+  | ArrayExpression
+  | TemplateLiteral
+
+export default function attrs({
```

Comment:
❓ I know that it's a draft and that we're probably going to move some things around. However, if this gets a bit further in the process, maybe we could add some tests on this for prosperity and self documentation, WYT?

Reply:
Definitely need tests, but I didn;t do it yet as it's a lot of work and I don't know if we'll be working on it yet.